sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -181,7 +182,7 @@ private T ConvertTo<T>()
         }
 
         /// <inheritdoc/>
-        public override string ToString()
+        public override string? ToString()
```

Comment:
~`ToString` should never return null.~

Apparently, .NET changed their minds on this: https://github.com/dotnet/runtime/issues/1318

Reply:
Yeah, I was surprised by that as well.  It's a super interesting debate, and potentially relevant to the conversation about [NullableResponseOfT](https://github.com/Azure/azure-sdk-for-net/pull/35560)'s nullable value field.

I think it's true, though, per [this comment](https://github.com/dotnet/coreclr/pull/23466#issuecomment-478112203) that we have a choice here about whether or not we want DynamicData to ever return null from `ToString()`.  We could prevent against it and then suppress any nullability warnings the compiler gave if we wanted to.

Reply:
I don't have a strong preference. I was just remembering ol' school rules. In practice, never seemed to really matter anyway. Whatever you, @KrzysztofCwalina, and @tg-msft think. :smile:

Reply:
My takeaway from the linked thread is that this override should not return nullable. Since we are the implementors, we can ensure it doesn't and based on guidance, it shouldn't. Is there a scenario here where we want null to be returned?

Reply:
Cool, I'll rework this so that the return type is not nullable.